docs/content/preview/yugabyte-voyager/known-issues/postgresql.md

```diff
@@ -1254,308 +1130,378 @@ EXECUTE FUNCTION check_and_modify_val();
+    LANGUAGE plpgsql
+    AS $$
+DECLARE
+    emp_salary employees.salary%TYPE;  -- Declare a variable with the same type as employees.salary
+BEGIN
+    SELECT salary INTO emp_salary
+    FROM employees
+    WHERE employee_id = emp_id;
+
+    RETURN emp_salary;
+END;
+$$;
 ```
 
-```output
-ERROR:  advisory locks feature is currently in preview
-HINT:  To enable this preview feature, set the GFlag ysql_yb_enable_advisory_locks to true and add it to the list of allowed preview flags i.e. GFlag allowed_preview_flags_csv. If the app doesn't need strict functionality, this error can be silenced by using the GFlag yb_silence_advisory_locks_not_supported_error. See https://github.com/yugabyte/yugabyte-db/issues/3642 for details
+Suggested change to CREATE FUNCTION is as follows:
+
+```sql
+CREATE FUNCTION public.get_employee_salary(emp_id integer) RETURNS numeric
+    LANGUAGE plpgsql
+    AS $$
+DECLARE
+    Emp_salary NUMERIC;  -- Declare a variable with the same type as employees.salary
+BEGIN
+    SELECT salary INTO emp_salary
+    FROM employees
+    WHERE employee_id = emp_id;
+
+    RETURN emp_salary;
+END;
+$$;
 ```
 
-**Workaround**: Implement a custom locking mechanism in the application to coordinate actions without relying on database-level advisory locks.
+---
+
+## Postgres 12 And later features
+
+### PostgreSQL 12 and later features
+
+**GitHub**: Issue [#25575](https://github.com/yugabyte/yugabyte-db/issues/25575)
+
+**Description**: If any of these PostgreSQL features for version 12 and later are present in the source schema, the import schema step on the target YugabyteDB will fail as YugabyteDB is currently PG11 compatible.
+
+- [JSON Constructor functions](https://www.postgresql.org/about/featurematrix/detail/395/) - `JSON_ARRAY_AGG`, `JSON_ARRAY`, `JSON_OBJECT`, `JSON_OBJECT_AGG`.
+- [JSON query functions](https://www.postgresql.org/docs/17/functions-json.html#FUNCTIONS-SQLJSON-TABLE) - `JSON_QUERY`, `JSON_VALUE`, `JSON_EXISTS`, `JSON_TABLE`.
+- [IS JSON predicate clause](https://www.postgresql.org/about/featurematrix/detail/396/).
+- Any Value [Aggregate function](https://www.postgresql.org/docs/16/functions-aggregate.html#id-1.5.8.27.5.2.4.1.1.1.1) - `any_value`.
+- [COPY FROM command with ON_ERROR](https://www.postgresql.org/about/featurematrix/detail/433/) option.
+- [Non-decimal integer literals](https://www.postgresql.org/about/featurematrix/detail/407/).
+- [Non-deterministic collations](https://www.postgresql.org/docs/12/collation.html#COLLATION-NONDETERMINISTIC).
+- [COMPRESSION clause](https://www.postgresql.org/docs/current/sql-createtable.html#SQL-CREATETABLE-PARMS-COMPRESSION) in TABLE Column for TOASTing method.
+- [CREATE DATABASE options](https://www.postgresql.org/docs/15/sql-createdatabase.html) (locale, collation, strategy, and oid related).
+
+Apart from these, the following issues are supported in YugabyteDB [v2.25](/preview/releases/ybdb-releases/v2.25), which supports PostgreSQL 15.
```

Comment:
I'm not sure what this means. Do you mean these will also fail, unless you are importing to 2.25+?

eg:
"In addition, if any of the following PostgreSQL features are present in the source schema, the import schema step on the target YugabyteDB will fail, unless you are importing to YugabyteDB v2.25 (which supports PG15).

Reply:
@makalaaneesh can you please help here?

Reply:
yes correct.  This makes sense.

> "In addition, if any of the following PostgreSQL features are present in the source schema, the import schema step on the target YugabyteDB will fail, unless you are importing to YugabyteDB v2.25 (which supports PG15).

Reply:
I've updated this sentence with what Dwight suggested.